rust/chains/abacus-ethereum/src/trait_builder.rs

```diff
@@ -0,0 +1,59 @@
+use std::sync::Arc;
```

Comment:
Lol this is wayyy more readable than the old macro

Reply:
Glad you think so. A little afraid raising this PR since it is technically a lot more lines of code, but I think easier to maintain ones personally.